oraclizeAPI_0.4.25.sol

```diff
@@ -366,155 +420,1286 @@ contract usingOraclize {
+            oraclize.addressCustomPaymentToken(address(this))
+        );
     }
 
-    function oraclize_getPrice(string datasource, uint gaslimit) oraclizeAPI internal returns (uint){
-        return oraclize.getPrice(datasource, gaslimit);
+    function oraclize_getPriceERC20(
+        string memory _datasource,
+        uint256 _gasLimit,
+        uint256 _gasPrice,
+        byte _proofType,
+        address _tokenAddress
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        _queryPrice = oraclize.price(
+            keccak256(
+                abi.encodePacked(
+                    _datasource,
+                    _proofType
+                )
+            )
+        );
+        return oraclize.convertToERC20Price(
+            _queryPrice + _gasLimit * _gasPrice,
+            _tokenAddress
+        );
     }
 
-    function oraclize_query(string datasource, string arg) oraclizeAPI internal returns (bytes32 id){
+    function oraclize_getPriceERC20(
+        bytes1 _datasource,
+        uint256 _gasLimit,
+        uint256 _gasPrice,
+        byte _proofType,
+        address _tokenAddress
+    )
+        oraclizeAPI
+        internal
+        returns (uint256 _queryPrice)
+    {
+        _queryPrice = oraclize.price(
+            keccak256(
+                abi.encodePacked(
+                    _datasource,
+                    _proofType
+                )
+            )
+        );
+        return oraclize.convertToERC20Price(
+            _queryPrice + _gasLimit * _gasPrice,
+            _tokenAddress
+        );
+    }
+    /**
+     *
+     * @notice `oraclize_query` overloads using the STRING datasource follow...
+     *
+     */
+    function oraclize_query(
+        string datasource,
+        string arg
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
         uint price = oraclize.getPrice(datasource);
-        if (price > 1 ether + tx.gasprice*200000) return 0; // unexpectedly high price
+        if (price > 1 ether + tx.gasprice * 200000) return 0; // unexpectedly high price
         return oraclize.query.value(price)(0, datasource, arg);
     }
-    function oraclize_query(uint timestamp, string datasource, string arg) oraclizeAPI internal returns (bytes32 id){
+
+    function oraclize_query(
+        uint timestamp,
+        string datasource,
+        string arg
+    )
+        oraclizeAPI
+        internal
+        returns (bytes32 id)
+    {
         uint price = oraclize.getPrice(datasource);
-        if (price > 1 ether + tx.gasprice*200000) return 0; // unexpectedly high price
+        if (price > 1 ether + tx.gasprice * 200000) return 0; // unexpectedly high price
         return oraclize.query.value(price)(timestamp, datasource, arg);
     }
-    function oraclize_query(uint timestamp, string datasource, string arg, uint gaslimit) oraclizeAPI internal returns (bytes32 id){
+
+    function oraclize_query(
+        uint timestamp,
```

Comment:
Pretty sure this would cost more gas since the EVM is optimized for 32 byte words and so more ops are needed to convert to/from when dealing with smaller sizes. Feel free to test.

Reply:
agree with greg, only makes sense for structs/storage.